mariadb_kernel/client_config.py

```diff
@@ -14,11 +14,20 @@ def __init__(self, log, name="mariadb_config.json"):
         self.default_config = {
             "user": "root",
             "host": "localhost",
+            "socket": "/tmp/mysqld.sock",
             "port": "3306",
             "password": "",
             "start_server": "True",
             "client_bin": "mysql",
             "server_bin": "mysqld",
+            "db_init_bin": "mysql_install_db",
+            "extra_server_config": ["--pid-file=/tmp/mysqld.pid",
+                                    f"--datadir={os.path.join(os.environ.get('HOME', '/home/jovyan/'), 'work', '.db')}",
+                                    "--skip_log_error"],
+            "extra_db_init_config": [f"--user={os.environ.get('NB_USER', 'jovyan')}",
```

Comment:
In my tests I need the user flag otherwise  mysql_install_db could not create the database files/folders. 
However, know when I think about it this might be a "follow up error" as user was previously configured in /etc/mysql/my.cnf as mysqld. I will test some more with --no-defaults as you suggested but currently --user is needed.

Reply:
Adding --no-defaults fix this so I will remove the user argument

Reply:
Amazing! 🎉